### PR TITLE
Update for Aff v4

### LINKED ISSUE #481: Needs Aff 4.0 upgrade
It seems like at least the change of Control.Monad.Fork -> Control.Monad.Fork.Class needs to be done first, and then there are some Aff-related changes that need to be done that I don't know how to do.

### PR DESCRIPTION
**Pull request by garyb**
Resolves #481

### CODE REVIEW

File: src/Halogen/Aff/Util.purs [hunk 1]
```diff
@@ -7,33 +7,31 @@ module Halogen.Aff.Util
...
-    >>= windowToEventTarget
-    >>> addEventListener load (eventListener (\_ -> callback unit)) false
+awaitLoad = makeAff \callback -> liftEff do
+  et <- windowToEventTarget <$> window
+  let listener = eventListener (\_ -> callback (Right unit))
+  addEventListener load listener false et
+  pure $ Canceler \_ -> liftEff (removeEventListener load listener false et)
 
 -- | Waits for the document to load and then finds the `body` element.
 awaitBody :: forall eff. Aff (dom :: DOM | eff) HTMLElement
```
Comment (natefaubion) [inline src/Halogen/Aff/Util.purs]: We should fix `awaitBody` so that it doesn't block forever if the document has already loaded.

Comment (garyb) [reply src/Halogen/Aff/Util.purs]: Ah yeah 👍

Comment (spicydonuts) [reply src/Halogen/Aff/Util.purs]: I took a stab at this here: https://github.com/garyb/purescript-halogen/pull/5

Comment (i-am-the-slime) [reply src/Halogen/Aff/Util.purs]: What's holding this back?

**Pull request merged**